ProcessLib/CompareJacobiansJacobianAssembler.cpp

```diff
@@ -0,0 +1,424 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2018, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#include "CompareJacobiansJacobianAssembler.h"
+
+#include <sstream>
+
+#include "MathLib/LinAlg/Eigen/EigenMapTools.h"
+
+#include "CreateJacobianAssembler.h"
+
+namespace
+{
+//! Dumps a \c double value as a Python script snippet.
+void dump_py(std::ofstream& fh, std::string const& var, double const val)
+{
+    fh << var << " = " << val << '\n';
+}
+
+//! Dumps a \c std::size_t value as a Python script snippet.
+void dump_py(std::ofstream& fh, std::string const& var, std::size_t const val)
+{
+    fh << var << " = " << val << '\n';
+}
+
+//! Dumps an Eigen vector as a Python script snippet.
+template <typename Vec>
+void dump_py_vec(std::ofstream& fh, std::string const& var, Vec const& val)
+{
+    fh << var << " = np.array([";
+    for (decltype(val.size()) i = 0; i < val.size(); ++i)
+    {
+        if (i != 0)
+        {
+            if (i % 8 == 0)
+            {
+                // Print at most eight entries on one line,
+                // indent with four spaces.
+                fh << ",\n    ";
+            }
+            else
+            {
+                fh << ", ";
+            }
+        }
+        fh << val[i];
+    }
+    fh << "])\n";
+}
+
+//! Dumps a \c std::vector<double> as a Python script snippet.
+void dump_py(std::ofstream& fh, std::string const& var,
+             std::vector<double> const& val)
+{
+    dump_py_vec(fh, var, val);
+}
+
+template <typename Derived>
```

Comment:
ah, ... and the Eigen vector comment goes here...

Reply:
:white_check_mark: